### PR TITLE
dotnet test output with color

### PR DESCRIPTION
**Pull request by smadala**
- Add color in missing place in vstest.console.exe
- VSTest task should use parent shell session.
- Add verbosity quiet option for console logger.

Related: https://github.com/Microsoft/vstest/issues/639

### CODE REVIEW

File: scripts/test.ps1 [hunk 1]
```diff
@@ -31,7 +31,11 @@ Param(
     [Switch] $FailFast = $false,
 
     [Parameter(Mandatory=$false)]
-    [Switch] $Parallel = $false
+    [Switch] $Parallel = $false,
+
+    [Parameter(Mandatory=$false)]
+    [Alias("d")]
+    [System.String] $ConsoleLogger = '/logger:"console;verbosity=minimal"'
```
Comment (AbhitejJohn) [inline scripts/test.ps1]: nit: Just make this take the arguments - "verbosity=minimal"

**Review by AbhitejJohn — APPROVED** [review]

File: scripts/test.ps1 (hunk 1 above)
Comment (codito) [inline scripts/test.ps1]: There is already a variable `-verbose`. Can we use that value and set this?

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 2]
```diff
@@ -218,15 +218,21 @@ private static void DisplayFullInformation(TestResult result)
                 hasData = true;
                 Output.WriteLine(CommandLineResources.ErrorMessageBanner, OutputLevel.Information);
                 string errorMessage = String.Format(CultureInfo.CurrentCulture, "{0}{1}", TestMessageFormattingPrefix, result.ErrorMessage);
-                Output.WriteLine(errorMessage, OutputLevel.Information);
+                using (new ConsoleColorHelper(ConsoleColor.Red))
+                {
+                    Output.WriteLine(errorMessage, OutputLevel.Error);
```
Comment (navin22) [inline src/vstest.console/Internal/ConsoleLogger.cs]: nit: how about creating a small method for this?

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs [hunk 3]
```diff
@@ -33,23 +33,15 @@ public int Execute()
                                       FileName = hostExe,
                                       Arguments = string.Join(" ", this.allArgs),
                                       UseShellExecute = false,
-                                      CreateNoWindow = true,
```
Comment (codito) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs]: What are the scenarios we have tried?

* `dotnet test` in cmd, powershell, bash
* `dotnet test` from Process.CreateProcess of any app
* Mimic how `vstest.console` is called from VSTS task or a jenkins/travis or any CI system

We should ensure that CI tasks that run as services (and thus not interactive) don't face any issues with this approach.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs [hunk 4]
```diff
@@ -33,23 +33,15 @@ public int Execute()
...
-                                      RedirectStandardError = true,
-                                      RedirectStandardOutput = true
                                   };
 
             Tracing.Trace("VSTest: Starting vstest.console...");
             Tracing.Trace("VSTest: Arguments: " + processInfo.FileName + " " + processInfo.Arguments);
 
             using (var activeProcess = new Process { StartInfo = processInfo })
             {
-                activeProcess.OutputDataReceived += (sender, args) => Console.WriteLine(args.Data);
```
Comment (codito) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs]: Good change 👍

File: src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs [hunk 5]
```diff
@@ -21,38 +21,44 @@ public static class OutputUtilities
...
 
         /// <summary>
         /// Output a informational message.
         /// </summary>
         /// <param name="output">Output instance the method is being invoked with.</param>
         /// <param name="format">Format string for the informational message.</param>
+        /// <param name="foregroundColor">Color in which text prints.</param>
         /// <param name="args">Arguments to format into the format string.</param>
-        public static void Information(this IOutput output, string format, params object[] args)
+        public static void Information(this IOutput output, string format, ConsoleColor foregroundColor=ConsoleColor.White, params object[] args)
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Output/OutputExtensions.cs]: We don't need to give the color for Error and Warning.
Probably, we can have overload for Information that takes in color ?

File: test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs [hunk 6]
```diff
@@ -21,42 +22,42 @@ public void TestInit()
         [TestMethod]
         public void OutputErrorForSimpleMessageShouldOutputTheMessageString()
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Output/OutputExtensionsTests.cs]: Add tests for verifying the color

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs (hunk 3 above)
Comment (shanselman) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs]: Also try "dotnet watch test" and make sure the color flows all the way through.

**Comment by shanselman** [comment]
Also try "dotnet watch test" and make sure the color flows all the way through.

File: src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs (hunk 3 above)
Comment (smadala) [inline src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingApp.cs]: @codito `dotnet test` working fine for above three scenarios. No color is displaying for `dotnet test` on VSTS task (same behavior as `dotnet build`).

@shanselman Yes, `dotnet watch test` displaying colored output.

**Review by codito — APPROVED** [review]

**Pull request merged**